config/routes.rb

```diff
@@ -1,6 +1,6 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
+  root to: "users#index"
+  get 'users/:id' => 'users#show', as: "user"
+  get 'users/:author_id/posts' => 'posts#index', as: "user_posts"
+  get 'users/:author_id/posts/:id' => 'posts#show', as: "user_post"
```

Comment:
- [ ] The routes with `author_id` should have `user_id` instead. We can also define these routes in a better way using Rails `resources` method.

  ```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
  ```